deno/fhirchat-demo/islands/ChatIsland.tsx

```diff
@@ -1,43 +1,54 @@
-import { useContext, useEffect } from "preact/hooks";
-import { AIConversationState } from "../hooks/aiConversationContext.ts";
+import { useContext, useEffect, useMemo } from "preact/hooks";
+import { signal } from "@preact/signals";
+import { AIConversationContext } from "../hooks/aiConversationContext.ts";
+import { Interaction, Message } from "../types/conversation.ts";
+import { extractInteractions } from "../helpers/extractConversationInteractions.ts";
+import UserMessage from "../components/UserMessage.tsx";
+import AgentMessage from "../components/AgentMessage.tsx";
 import { useTextWithTypeAnimation } from "../hooks/typingAnimation.ts";
 
 export default function ChatIsland() {
   const {
-    question,
     conversation,
-    setQuestion,
-    appendToConversation,
-    submitQuestion,
     closeConversation,
-  } = useContext(AIConversationState);
+    submitQuestion,
+  } = useContext(AIConversationContext);
 
-  const userQuestion = conversation.value[0];
-  const agentResponse = conversation.value[1];
+  const interactions: Array<Interaction> = useMemo(() =>
+    extractInteractions(
+      conversation.value,
+    ), [conversation.value]);
+
+  const userInput = signal<string>("");
+  const mostRecentAgentResponse = interactions.at(-1)?.agentMessage;
 
   const { displayText, resetTextAnimation } = useTextWithTypeAnimation({
-    text: agentResponse,
-    enabled: agentResponse !== "",
+    text: mostRecentAgentResponse?.message,
+    enabled: !!mostRecentAgentResponse?.message,
   });
 
+  const determineAgentDisplayText = (agentMessage?: Message) => {
+    return agentMessage?.id === mostRecentAgentResponse?.id
+      ? displayText
+      : agentMessage?.message ?? "";
```

Comment:
nit: can we remove a `agentMessage?.message ?? ""` to its own const, otherwise it looks like a nested ternary.

Reply:
Sounds good!